File: Objective-C Playground/Objective-C Playground/User Interface/MainView.m
#import "MainView.h"

#pragma mark - Interface

@interface MainView()

@property UILabel *label;

@end

#pragma mark - Implementation

@implementation MainView

UILabel *label;

- (id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    
    if (self) {
        [self setupSelf];
    }
    
    return self;
}

- (void)setupSelf {
    self.backgroundColor = UIColor.systemBackgroundColor;
    [self setupSubviews];
    [self setupConstraints];
}

- (void)setupSubviews {
    [self setupLabel];
}

- (void)setupLabel {
    self.label = [[UILabel alloc] initWithFrame:CGRectZero];
    self.label.translatesAutoresizingMaskIntoConstraints = NO;
    self.label.text = @"Hello, world!";
    [self addSubview:self.label];
}

- (void)setupConstraints {
    [NSLayoutConstraint activateConstraints:@[
        [self.label.centerXAnchor constraintEqualToAnchor:self.centerXAnchor],
        [self.label.centerYAnchor constraintEqualToAnchor:self.centerYAnchor],
    ]];
}

@end

File: Programmatic UIKit Playground/Programmatic UIKit Playground/User Interface/MainView.swift
import UIKit

class MainView: UIView {
    private let label = UILabel(frame: .zero)

    override init(frame: CGRect) {
        super.init(frame: frame)
        setupSelf()
    }

    @available(*, unavailable) required init?(coder: NSCoder) { fatalError() }

    private func setupSelf() {
        backgroundColor = .systemBackground
        setupSubviews()
        setupConstraints()
    }

    private func setupSubviews() {
        setupLabel()
    }

    private func setupLabel() {
        label.translatesAutoresizingMaskIntoConstraints = false
        label.text = "Hello, world!"
        addSubview(label)
    }

    private func setupConstraints() {
        NSLayoutConstraint.activate([
            label.centerXAnchor.constraint(equalTo: centerXAnchor),
            label.centerYAnchor.constraint(equalTo: centerYAnchor),
        ])
    }
}
